File: LeetCode/problems/p0027/python/p027.py
#[COMPLETED]

class Solution(object):
    """
    :type nums: List[int]
    :type val: int
    :rtype: int
    """
    def removeElement(self, nums, val):

        num_removed = 0
        for i,n in enumerate(nums):
            if n == val:
                num_removed += 1
            else:
                nums[i - num_removed] = nums[i]
        return len(nums) - num_removed


sol = Solution()
l = [3,2,2,3]
print(sol.removeElement(l, 3), "expected 2")

l = [0,1,2,2,3,0,4,2]
print(sol.removeElement(l, 2), "expected 5")

File: leet-code/src/problems/p0027/java/src/p027/java/P027.java
//[COMPLETED]

package p027;

import mathtools.MF;

public class P027 {

    public static void main(String[] args) {
        int[] arr = new int[] {};
        Solution s = new Solution();
        
        System.out.println(s.removeElement(arr, 2));
        
        MF.printList(arr);
        
    }

    
private static class Solution {
    public int removeElement(int[] nums, int val) {
        
        //if (nums == null || nums.length == 0)
        //    return 0;
        
        int j = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == val)
                continue;
            
            nums[j] = nums[i];
            j++;
        }
        
        //dont need
        for (int i = j; i < nums.length; i++)
            nums[i] = 0;
        
        return j;
    }
}

}
